src/EditorFeatures/Core/Implementation/CodeCleanup/CodeCleanupService.cs

```diff
@@ -0,0 +1,165 @@
+            dictionary.Add(FeatureOnOffOptions.FixAccessibilityModifiers,
+                new[] { IDEDiagnosticIds.AddAccessibilityModifiersDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.SortAccessibilityModifiers,
+                new[] { IDEDiagnosticIds.OrderModifiersDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.MakeReadonly,
+                new[] { IDEDiagnosticIds.MakeFieldReadonlyDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.RemoveUnnecessaryCasts,
+                new[] { IDEDiagnosticIds.RemoveUnnecessaryCastDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixExpressionBodiedMembers,
+                new[] { IDEDiagnosticIds.UseExpressionBodyForConstructorsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForMethodsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForConversionOperatorsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForOperatorsDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForPropertiesDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixInlineVariableDeclarations,
+                new[] { IDEDiagnosticIds.InlineDeclarationDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.RemoveUnusedVariables,
+                new[] { "CS0168", "CS0219" });
+            dictionary.Add(FeatureOnOffOptions.FixObjectCollectionInitialization,
+                new[] { IDEDiagnosticIds.UseObjectInitializerDiagnosticId, IDEDiagnosticIds.UseCollectionInitializerDiagnosticId });
+            //dictionary.Add(FeatureOnOffOptions.FixLanguageFeatures,
+            //    new[] { IDEDiagnosticIds. });
+            //IDEDiagnosticIds.PreferIntrinsicPredefinedTypeInDeclarationsDiagnosticId,
+            //    IDEDiagnosticIds.PreferIntrinsicPredefinedTypeInMemberAccessDiagnosticId,
+            //    IDEDiagnosticIds.InlineAsTypeCheckId,
+            //    IDEDiagnosticIds.InlineIsTypeCheckId,
+            //    IDEDiagnosticIds.InlineIsTypeWithoutNameCheckDiagnosticsId,
+
+            return dictionary.ToImmutableDictionary();
+        }
+
+        public Task<IEnumerable<TextChange>> GetChangesForCleanupDocument(Document document, CancellationToken cancellationToken)
+        {
+            var oldDocument = document;
+            document = RemoveSortUsings(document, cancellationToken);
+            document = ApplyCodeFixes(document, cancellationToken);
+
+            return document.GetTextChangesAsync(oldDocument, cancellationToken);
+        }
+        private Document RemoveSortUsings(Document document, CancellationToken cancellationToken)
+        {
+            // remove and sort usings
+            if (document.Project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.RemoveUnusedUsings, LanguageNames.CSharp))
```

Comment:
this is in a language neutral layer, but you're referring to LanguageNames.CSharp here.  This seems wrong.

Reply:
fixed